### PR TITLE
[SYCL] Add test for zero size range kernels.

### PR DESCRIPTION
**Pull request by mmoadeli**
Adds test for zero size range kernel for [PR 7044](https://github.com/intel/llvm/pull/7044).

### CODE REVIEW

File: SYCL/Basic/zero_size_range.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
```
Comment (againull) [inline SYCL/Basic/zero_size_range.cpp:2]: This line only builds the test. So, currently test it is not executed.
Could you please add lines to execute the test?

File: SYCL/Basic/zero_size_range.cpp [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %t.out
```
Comment (againull) [inline SYCL/Basic/zero_size_range.cpp]: This line will execute the test only on device/bakcend which is chosen by default, for example gpu with l0 backend.
I guess you need to use placeholders to correctly run this test on all devices of interest. Please take a look at other tests, for example: https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Basic/buffer/buffer_full_copy.cpp

**Review by againull — APPROVED** [review]

**Comment by againull** [comment]
It looks like test fails on Level Zero and Opencl backends - http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLLVM_Test_Suite_Associate_CI/detail/LLVM_Test_Suite_Associate_CI/375/pipeline

**Comment by mmoadeli** [comment]
> It looks like test fails on Level Zero and Opencl backends - http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLLVM_Test_Suite_Associate_CI/detail/LLVM_Test_Suite_Associate_CI/375/pipeline

It should be tested using https://github.com/intel/llvm/pull/7044

**Comment by againull** [comment]
> > It looks like test fails on Level Zero and Opencl backends - http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLLVM_Test_Suite_Associate_CI/detail/LLVM_Test_Suite_Associate_CI/375/pipeline
> 
> It should be tested using [intel/llvm#7044](https://github.com/intel/llvm/pull/7044)

It is tested using this PR. It was triggered by commend: "/verify with https://github.com/intel/llvm/pull/7044"
The following Jenkins job corresponds to such associated testing: Jenkins/llvm-test-suite

**Comment by againull** [comment]
> > > It looks like test fails on Level Zero and Opencl backends - http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLLVM_Test_Suite_Associate_CI/detail/LLVM_Test_Suite_Associate_CI/375/pipeline
> > 
> > 
> > It should be tested using [intel/llvm#7044](https://github.com/intel/llvm/pull/7044)
> 
> It is tested using this PR. It was triggered by commend: "/verify with [intel/llvm#7044](https://github.com/intel/llvm/pull/7044)" The following Jenkins job corresponds to such associated testing: Jenkins/llvm-test-suite

It looks like your PR fixes the problem for CUDA and HIP backends. Is this test expected to pass for L0 and OpenCL? Should execution be limited only for CUDA and HIP backends?

**Comment by mmoadeli** [comment]
> > > > It looks like test fails on Level Zero and Opencl backends - http://icl-jenkins2.sc.intel.com:8080/blue/organizations/jenkins/SYCL_CI%2Fintel%2FLLVM_Test_Suite_Associate_CI/detail/LLVM_Test_Suite_Associate_CI/375/pipeline
> > > 
> > > 
> > > It should be tested using [intel/llvm#7044](https://github.com/intel/llvm/pull/7044)
> > 
> > 
> > It is tested using this PR. It was triggered by commend: "/verify with [intel/llvm#7044](https://github.com/intel/llvm/pull/7044)" The following Jenkins job corresponds to such associated testing: Jenkins/llvm-test-suite
> 
> It looks like your PR fixes the problem for CUDA and HIP backends. Is this test expected to pass for L0 and OpenCL? Should execution be limited only for CUDA and HIP backends?

The initial fix was intended for cuda and hip backends only. I successfully tried it with opencl:cpu (using ONEAPI_DEVICE_SELECTOR=opencl:cpu) and cuda backends locally. 
Given: 
* running CI on opencl and level zero shows failure, and  
* the focus of work has been addressing cuda and hip, 

I limit execution for cuda and hip backends and create an issue to address the failures seen on opencl and level zero.
